### ISSUE TITLE
Update testnet generation logic for actual deployment scenarios

### ISSUE THREAD

**Issue by conorsch** [penumbra-zone/penumbra#2882]
We've long had support for setting ip information during creation of a new testnet, via `pd testnet generate --starting-ip <ip>`. We don't use that logic when deploying testnets, however, because it hardcodes adjacent IPs that don't map to our (or any) cloud environment. Instead, we clobber the generated testnet info via bash: https://github.com/penumbra-zone/penumbra/blob/189690fe73e4eae46f341a7d0989bcc16f4fe1ea/deployments/ci.sh#L63-L92

We should permit runtime overrides for at least one of the following:

* `--peer-address-template`, with e.g. `penumbra-testnet-p2p-val` passed in during deploys, so that `-0` and `-1` are appended automatically, one per validator. Doing so would support private access to genesis validators via their service endpoints, but not require us to expose the genesis validators to external ingress.
* `--external-address-template`, a similar option, permitting external ingress access directly to genesis validators, suitable for devnets (where we might not want to set up a load-balanced service layer of fullnodes to handle external client requests).

Making these changes would allow us to be more flexible in the type and number of networks that we deploy, supporting e.g. custom networks to evaluate upgrade-via-governance behavior as described in #1804. 

While we're at it, there's a fair amount of refactoring of the tall functions in the `pd::testnet::*` modules that would benefit from more types with to/from impls. #679 is within reach during such a refactor, as is #1727, but that's more of a nice-to-have. #1832 would be resolved automatically by these changes, and in combination with #2858, we could finally address #1783 meaningfully.

That's a lot of cross-linking, but the core ask here is that running `pd testnet generate` actually emits usable information for our test deployments. Otherwise, we're needless rewriting testnet generation in bash, rather than rust.

**Comment by conorsch** [comment]
Added in #2916. The CI logic still clobbers the emitted info, but I'll work on updating the CI logic opportunistically. That work is now unblocked, thanks to the changes in 2916, which was the goal of this ticket.
